### DISCUSSION TITLE
Is this library working for ESP8266 (NodeMCU)?

### DISCUSSION THREAD

**Discussion by imargonis** [pschatzmann/arduino-audio-tools#362] [Q&A] [answered]
Hello,
I am trying to use this library with an ESP8266-12F (NodeMCU) but I have some problems.

In particular I am interested in recording from an analog microphone but I cannot even get the streams-adc-serial example to compile. The class 'AnalogAudioStream' is not defined (probably because USE_ADC_ARDUINO is not defined).

Is this a bug or is the ESP8266 not supported (at least for analog input)?

**Comment by pschatzmann** [comment, marked as answer]
The ESP32 provides analog input with the help of of the I2S interface using DMA.
The ESP8266 does not have this functionality: so for the time being AnalogAudioStream is not supported.
And looking at the restrictions of the ESP8266 ADC it's not worth the effort.

**Comment by imargonis** [comment]
Oh, I did not expect such a quick answer! 
Although it is not what I hoped for, thank you very much.
I guess I will have to look for other solutions (maybe using an external ADC and using a digital input would work? I am still a novice so maybe I am talking nonsense here).

**Comment by pschatzmann** [comment]
A digital microphone should work: see https://github.com/pschatzmann/arduino-audio-tools/wiki/External-ADC
